### PR TITLE
[#3983] refactor: move the schema operation ITs to FlinkCommonIT

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

refactor: move the schema operation ITs to FlinkCommonIT

### Why are the changes needed?

Fix: #3983 

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

 - ITs

### CODE REVIEW

**Comment by coolderli** [comment]
@FANNG1 I separate the Flink connector MR. This is some refactor. Please take a look. Thanks.

**Comment by FANNG1** [comment]
> @FANNG1 I separate the Flink connector MR. This is some refactor. Please take a look. Thanks.

Suggest to refer SparkIT,  such as schema tests place in `FlinkcommonIT` not in `FlinkHiveSchemaIT` is more reasonable.

File: flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java [hunk 1]
```diff
@@ -64,9 +64,11 @@ public static GravitinoCatalogManager get() {
    * <p>After close, GravitinoCatalogManager can not be used anymore.
    */
   public void close() {
-    Preconditions.checkState(!isClosed, "Gravitino Catalog is already closed");
-    isClosed = true;
-    gravitinoClient.close();
+    if (!isClosed) {
```
Comment (FANNG1) [inline flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java:67]: why changing this?

**Comment by FANNG1** [comment]
> > > @FANNG1 I separate the Flink connector MR. This is some refactor. Please take a look. Thanks.
> > 
> > 
> > Suggest to refer SparkIT, such as schema tests place in `FlinkcommonIT` not in `FlinkHiveSchemaIT` is more reasonable.
> 
> @FANNG1 I finished it. I want to use parameterize to test schema. Please take a look. Thanks.

Junit 5 seems doesn't support class level parameter tests, it just supports method level

**Comment by coolderli** [comment]
> > > > @FANNG1 I separate the Flink connector MR. This is some refactor. Please take a look. Thanks.
> > > 
> > > 
> > > Suggest to refer SparkIT, such as schema tests place in `FlinkcommonIT` not in `FlinkHiveSchemaIT` is more reasonable.
> > 
> > 
> > @FANNG1 I finished it. I want to use parameterize to test schema. Please take a look. Thanks.
> 
> Junit 5 seems doesn't support class level parameter tests, it just supports method level

@FANNG1 I think the method parameter test is enough. Please take a look.

**Comment by coolderli** [comment]
@FANNG1 Please help review this. Thanks.

File: flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java (hunk 1 above)
Comment (FANNG1) [reply flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java:67]: Why `gravitinoCatalogManager` may be closed and recreated again?

**Comment by FANNG1** [comment]
LGTM, except one comment

File: flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java (hunk 1 above)
Comment (coolderli) [reply flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java:67]: @FANNG1 The gravitinoCatalogManager is managed by CatalogStore and CatalogStore will always be closed when invoking the `stop()` in FlinkEnvIT. When multiple tests inherit the FlinnkEnvIT, the gravitinoCatalogManager will be closed and recreated.

Comment (FANNG1) [reply flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java:67]: got it

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@coolderli , merged to main, thanks for your work!

**Comment by FANNG1** [comment]
@coolderli ,  `@Tag("gravitino-docker-it")` should be corrected to `@Tag("gravitino-docker-test")`,  could you propose a PR to fix it?

**Pull request merged**